v0/destinations/am/transform.js

```diff
@@ -105,18 +91,80 @@ function getSessionId(payload) {
 function addMinIdlength() {
   return { min_id_length: 1 };
 }
+function createRevenuePayload(
+  destination,
+  message,
+  rawPayload,
+  numberOfEvents,
+  revenueTypeCalculation
+) {
+  if (destination.Config.trackRevenuePerProduct) {
+    // In this case per product revenue is calculated from each product purchased event
+    // also when there is no product array the revenue will be calculated from the very first event only
+    // AM calculates revenue when revenue, price or quantity is specified on the roots but not inside event_properties
+    if (message.event === "Product Purchased" || numberOfEvents === 1) {
+      if (message.properties.revenue)
+        rawPayload.revenue = message.properties.revenue;
+      if (message.properties.price) {
+        rawPayload.price = message.properties.price;
+      }
+      if (message.properties.quantity) {
+        rawPayload.quantity = message.properties.quantity;
+      }
+      if (message.event === "Product Purchased") {
+        // here the revenue of the root is sent through revenueTypeCalculation because
+        // product purchased event does not have this field
+        if (revenueTypeCalculation >= 0) {
```

Comment:
we should consider price/revenue of each product to determine revenueType for product purchased event